### PR TITLE
Creates unittests for addrman, makes addrman more testable.

### PR DESCRIPTION
**Pull request by eth0**
- Creates unittests for addrman, while not all addrman behavior is covered by these tests they provide a useful starting place and increase code coverage.
- Minor modifications to addrman to allow deterministic and effective testing.

### CODE REVIEW

File: src/addrman.h [hunk 1]
```diff
@@ -567,6 +569,14 @@ class CAddrMan
             Check();
         }
     }
+
+
+    #ifdef __ADDRMAN_TEST__
```
Comment (jonasschnelli) [inline src/addrman.h]: Would it make sense to link this to the configure flag `--enable-tests`?

**Comment by eth0** [comment]
@jonasschnelli How would one learn the value of --enable-tests from a macro level? I suppose you could move this to its own source file and then only include that source when building tests, but that is a bit complicated.

What advantages are offered by using the --enable-tests flag?

**Comment by jonasschnelli** [comment]
@eth0: maybe remote the `#ifdef __ADDRMAN_TEST__` completely or export the `ENABLE_TESTS` over `AC_DEFINE_UNQUOTED()` in `configure.ac` and use this define to determine if you compile the `MakeDeterministic()` function?

Is the `newOnly` related to the new unit-tests?

Nice work!
Concept ACK.

**Comment by eth0** [comment]
@jonasschnelli  
I removed the macro completely. 

```newOnly``` is related to the unit-tests since I need a way to test if an addr is in the new or tried tables. 

I am developing some non-unittest code which builds on the ```newOnly``` functionality but it is independent of this pull-request.

**Comment by dcousens** [comment]
concept ACK,  non test-code utACK

**Comment by EthanHeilman** [comment]
@dcousens Anything I can do to move this along? 
I've tested this code in the Bitcoin client and the unittests all pass.

**Comment by jgarzik** [comment]
ut ACK - doesn't test very much, but what it does test, looks correct based on quick review :)

**Comment by laanwj** [comment]
utACK

**Pull request merged**